pan-domain-node/src/panda.ts

```diff
@@ -0,0 +1,111 @@
+import * as iniparser from 'iniparser';
+import * as cookie from 'cookie';
+
+import { httpGet, base64ToPEM, parseCookie, verifySignature, parseUser } from './utils';
+import { ValidateUserFn, AuthenticationStatus, User, AuthenticationResult } from './api';
+
+interface PublicKeyHolder {
+    key: string,
+    lastUpdated: Date
+}
+
+function fetchPublicKey(region: string, bucket: String, keyFile: String): Promise<PublicKeyHolder> {
+    const path = `https://s3.${region}.amazonaws.com/${bucket}/${keyFile}`;
+
+    return httpGet(path).then(response => {
+        const config: { publicKey?: string} = iniparser.parseString(response);
+        
+        if(config.publicKey) {
+            return {
+                key: base64ToPEM(config.publicKey),
+                lastUpdated: new Date()
+            };
+        } else {
+            throw new Error("Missing publicKey setting from config");
+        }
+    });
+}
+
+export function verifyUser(pandaCookie: string | undefined, publicKey: string, now: number, validateUser: ValidateUserFn): AuthenticationResult {
```

Comment:
`now` isn't an especially clearly named variable, that is, `now` doesn't lend it self to be assignable

Reply:
I've renamed it to `currentTimestamp`